dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/RequestPriorityAggregator.kt

```diff
@@ -0,0 +1,106 @@
+package org.dataland.datalandcommunitymanager.services
+
+import org.dataland.datalandbackend.openApiClient.model.DataTypeEnum
+import org.dataland.datalandcommunitymanager.model.dataRequest.AggregatedDataRequest
+import org.dataland.datalandcommunitymanager.model.dataRequest.AggregatedDataRequestWithAggregatedPriority
+import org.dataland.datalandcommunitymanager.model.dataRequest.AggregatedRequestPriority
+import org.dataland.datalandcommunitymanager.model.dataRequest.RequestPriority
+
+/**
+ * Implementation of a request priority aggregator to determine the aggregated priority
+ * and filter for a certain aggregated priority
+ */
+class RequestPriorityAggregator {
+    /**
+     * This method determines the aggregated priorities of aggregated data requests.
+     * @param aggregatedDataRequests the aggregated data requests
+     * @returns aggregated data requests with their aggregated priority
+     */
+    fun aggregateRequestPriority(aggregatedDataRequests: List<AggregatedDataRequest>): List<AggregatedDataRequestWithAggregatedPriority> {
+        val aggregatedMap = createAggregatedMap(aggregatedDataRequests)
+        return aggregatedMap
+            .map { (key, priorityCounts) ->
```

Comment:
```suggestion
            .map { (key, priorityCount) ->
```
should be singular as there is only one count for each key, right?

Reply:
priorityCountMap :)